### DISCUSSION TITLE
Does prisma support custom value transformers?

### DISCUSSION THREAD

**Discussion by JamesDay127** [prisma/prisma#14994] [Q&A] [answered]
Hi all,
The company I work for is investigating moving from Typeorm to Prisma. We just have one sticking point that hopefully we can find an answer for, then we're good to go!

We currently use the `transformer` method a lot in our Typeorm models and I haven't seen the same functionality in Prisma. Have I missed it, or is there a way to implement the same thing in the Prisma model?

For example:
A legacy field is a unix timestamp integer, and we have a custom transformer on the model to interact with it as a Date object on read/write. 

Thanks in advance for any advice

**Comment by austincrim** [comment, marked as answer]
Hey @JamesDay127 👋,

You could look into [Prisma middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) as a potential solution for this. 

We are also working on a more advanced solution in Prisma Client extensions, so stay tuned for that 🤞.

**Reply by JamesDay127** [reply]
Thanks @austincrim, I'll check out the middleware approach. 

And thanks for the update, looking forward to the extension!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
